website/static/js/components/quickFiles.js

```diff
@@ -0,0 +1,150 @@
+        method : 'GET',
+        url : url,
+        background : true,
+        config: mHelpers.apiV2Config({withCredentials: window.contextVars.isOnRootDomain})
+    });
+
+    promise.then(
+        function(result) {
+            ctrl.requestPending(false);
+            ctrl.quickFiles(result.data);
+            updatePagination(result, url);
+            m.redraw();
+            return promise;
+        }, function(xhr, textStatus, error) {
+            ctrl.failed = true;
+            ctrl.requestPending(false);
+            m.redraw();
+            Raven.captureMessage('Error retrieving quickfiles', {
+                extra: {
+                    url: url,
+                    textStatus: textStatus,
+                    error: error
+                }
+            });
+        }
+    );
+};
+
+
+var QuickFile = {
+
+    controller: function(options) {
+        var self = this;
+        self.file = options.file;
+        self.icon =  iconmap.file;
+    },
+
+    view: function(ctrl)  {
+        var viewBase = window.location.origin + '/quickfiles';
+        var viewUrl = ctrl.file.attributes.guid ? viewBase + '/' + ctrl.file.attributes.guid : viewBase + ctrl.file.attributes.path;
+        return m('div', [
+            m('li.project list-group-item list-group-item-node cite-container', [
+                m('h4.list-group-item-heading', [
+                    m('span.component-overflow.f-w-lg', {style: 'line-height: 1.5; width: 100%'}, [
+                        m('span.project-statuses-lg', {style: 'width: 50%; float:left'}, [
+                            m('span', {class: ctrl.icon, style: 'padding-right: 5px;'}, ''),
+                            m('a', {'href': viewUrl,
+                                onclick : function () {
+                                    $osf.trackClick('QuickFiles', 'view', 'view-quickfile-from-profile-page');
+                                }
+                            }, ctrl.file.attributes.name),
+                        ]),
+                        m('div', {style: 'width: 50%; float:right; font-size:small; line-height:2;'}, 'downloads: ' + ctrl.file.attributes.extra.downloads)
+                    ])
+                ])
+            ])
+        ]);
+    }
+};
+
+var QuickFiles = {
+
+    controller: function (options) {
+        var self = this;
+        self.user = options.user._id;
+        self.isProfile = options.user.is_profile;
+
+        self.quickFiles = m.prop([]);
+        self.requestPending = m.prop(false);
+
+        self.getCurrentQuickFiles = function _getCurrentQuickFiles(page) {
+            if (!self.requestPending()) {
+                var url = _buildUrl(page, self.user);
+                return _getNextItems(self, url, options.updatePagination);
+            }
+        };
+        self.getCurrentQuickFiles();
+    },
+
+    view: function (ctrl) {
+
+        return m('ul.list-group m-md', [
+            // Error message if the request fails
+            ctrl.failed ? m('p', [
```

Comment:
`ctrl.failed` should initialized in the controller.

Reply:
good call! initialized in controller for both this and public nodes!